.github/workflows/auto.yaml

```diff
@@ -0,0 +1,121 @@
+name: GitHub CI merge checks
+
+on:
+  push:
+    branches: [ "auto" ]
+
+env:
+  CHECKOUT_FETCH_DEPTH: 1001
+
+jobs:
+  linux-distros-tests:
+
+    strategy:
+      matrix:
+        os:
+          - centos-stream-9
+          - debian-stable
+          - debian-testing
+          - debian-unstable
+          - fedora-39
+          - fedora-40
+          - fedora-rawhide
+          - gentoo
+          - opensuse-leap
+          - opensuse-tumbleweed
+          - ubuntu-focal
+          - ubuntu-jammy
+          - ubuntu-noble # EOL 2036-04
+          - ubuntu-oracular # EOL 2025-07
+        compiler:
+          - { CC: gcc, CXX: g++ }
+          - { CC: clang, CXX: clang++ }
+        layer:
+          - { name: layer-00-default, nick: default }
+          - { name: layer-01-minimal, nick: minimal }
+          - { name: layer-02-maximus, nick: maximus }
+
+    runs-on: ubuntu-22.04
+    container:
+      image: squidcache/buildfarm-${{ matrix.os }}:stable
+      options: --user 1001 # uid used by worfklow runner
+
+    name: linux-distros-tests(${{ matrix.os }},${{ matrix.compiler.CC }},${{ matrix.layer.nick }})
+    env:
+      CC: ${{ matrix.compiler.CC }}
+      CXX: ${{ matrix.compiler.CXX }}
+
+    steps:
+      - name: Checkout Sources
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: ${{ env.CHECKOUT_FETCH_DEPTH }}
```

Comment:
If these two lines were added by accident, please remove them. AFAICT, they do not exist in the relevant parts of the official sources (i.e. in default.yaml instructions for `build-tests` jobs[^1]), and they force [actions/checkout](https://github.com/actions/checkout?tab=readme-ov-file#checkout-v4) to do more work than necessary for our build tests: "Only a single commit is fetched _by default_".

The same concern applies to MacOS tests, of course -- this part of the code is duplicated in the current PR.

If all extra depth configurations is removed from this file, please remove `CHECKOUT_FETCH_DEPTH` declaration as well (as unused).

[^1]: Fetching extra Squid repository commits is necessary for our `source-maintenance-tests`, but this new workflow does not run those tests.

Reply:
Got you. I was wandering in fact; the code that uses part of squid's commits is not in the github actions but in the scripts.
It wasn't clear to me in the comments in the current master code.
Removed.